### PR TITLE
Update listmonk@.service

### PR DESCRIPTION
**Pull request by MaximilianKohler**
https://github.com/knadh/listmonk/issues/843#issuecomment-1836783468

### CODE REVIEW

**Comment by MaximilianKohler** [comment]
> Your commit breaks the [service template unit](https://www.freedesktop.org/software/systemd/man/latest/systemd.unit.html). That `%i` is there for a reason. Please try to understand what you modify.

Are you talking about `ConditionPathExists=/etc/listmonk/%i.toml` ? Per the referenced issue https://github.com/knadh/listmonk/issues/843 it causes an error, so [this guide](https://mumaritc.hashnode.dev/how-to-install-listmonk-using-binary-on-ubuntu-2204), discussed in that issue, recommends changing it. 

I didn't try without changing those lines since that person had an issue + the guide says to change them, but it worked according to the guide, so that's why I suggested changing it.

**Comment by MaximilianKohler** [comment]
Good to know. I tagged you in the issue because I assumed you know more about this than me. So please feel free to make specific suggestions/changes.

**Comment by MaximilianKohler** [comment]
Apparently my listmonk service stopped on its own:
```
systemctl status listmonk
● listmonk.service
   Loaded: loaded (/etc/systemd/system/listmonk.service; enabled; vendor preset: disabled)
   Active: failed (Result: start-limit) since Sat 2023-12-09 21:18:45 PST; 23h ago

Dec 09 21:31:37 domain systemd[1]: [/etc/systemd/system/listmonk.service:24] Not an absolute path, ignoring: %S/listmonk-
Dec 09 21:31:37 domain systemd[1]: [/etc/systemd/system/listmonk.service:47] Unknown lvalue 'ProtectControlGroups' in s...ervice'
Dec 09 21:31:37 domain systemd[1]: [/etc/systemd/system/listmonk.service:48] Unknown lvalue 'ProtectKernelTunables' in ...ervice'
Dec 09 21:31:37 domain systemd[1]: [/etc/systemd/system/listmonk.service:56] Invalid device node path 'False'. Ignoring.
Dec 09 21:31:37 domain systemd[1]: [/etc/systemd/system/listmonk.service:58] Unknown lvalue 'RestrictNamespaces' in sec...ervice'
Dec 09 21:31:37 domain systemd[1]: [/etc/systemd/system/listmonk.service:60] Unknown lvalue 'RestrictRealtime' in secti...ervice'
Dec 09 21:31:37 domain systemd[1]: [/etc/systemd/system/listmonk.service:65] Unknown lvalue 'MemoryDenyWriteExecute' in...ervice'
Dec 09 21:31:37 domain systemd[1]: [/etc/systemd/system/listmonk.service:67] Unknown lvalue 'LockPersonality' in sectio...ervice'
Dec 09 21:31:37 domain systemd[1]: [/etc/systemd/system/listmonk.service:71] Unknown lvalue 'ProtectKernelModules' in s...ervice'
Dec 09 21:31:37 domain systemd[1]: [/etc/systemd/system/listmonk.service:74] Unknown lvalue 'PrivateUsers' in section 'Service'
Hint: Some lines were ellipsized, use -l to show in full.
```

Is it possible to send an email notification when that happens? I know my CSF firewall is able to email me about events. I modified these lines to try to fix the issue:
I commented out the 2nd line here:
```
EnvironmentFile=-/etc/default/listmonk
# EnvironmentFile=-/etc/default/listmonk-%i
```
I'm not sure whether to comment out these lines:
```
StateDirectory=listmonk-%i
Environment=HOME=%S/listmonk-%i
WorkingDirectory=%S/listmonk-%i
```
Or change them to:
```
StateDirectory=/etc/listmonk
Environment=HOME=/etc/listmonk
WorkingDirectory=/usr/bin
```
I was thinking to [simplify it the same way bamboowonder did](https://github.com/knadh/listmonk/issues/843#issuecomment-1159822295), but that guide doesn't cover creating a new user for `User=<yourUser>`. I saw another page showing that you can run it without specifying a user, so for now I've done that.

However, as noted here https://github.com/knadh/listmonk/issues/843#issuecomment-1836023524 another way to "fix this" is by running `systemctl restart listmonk` after `systemctl daemon-reload`. Is there a way to make it do that automatically?

Oh, I think it didn't restart after the server rebooted. So this needs to be changed too:
`Restart=on-failure` to `Restart=always` or something else: https://www.freedesktop.org/software/systemd/man/latest/systemd.service.html

## Summary:

This file needs major changes to be more compatible. I suggested in the issue-thread that we could have two versions or two sets of commands in the same file with one set commented out by default.

Perhaps we could leave the current file mostly the same _(and/or rename it to `listmonk-multi@.service`)_ and create a new `listmonk-simple@.service` file containing:
```
[Unit]
Description=listmonk email service
ConditionPathExists=/etc/listmonk/config.toml
Wants=network.target
# The PostgreSQL database may not be on the same host but if it
# is listmonk should wait for it to start up.
After=postgresql.service

[Service]
Type=simple
PermissionsStartOnly=true
WorkingDirectory=/usr/bin
EnvironmentFile=-/etc/default/listmonk
ExecStartPre=/usr/bin/mkdir -p "/listmonk/uploads"
ExecStartPre=/usr/bin/listmonk --config /etc/listmonk/config.toml --upgrade --yes
ExecStart=/usr/bin/listmonk --config /etc/listmonk/config.toml $SYSTEMD_LISTMONK_ARGS
Restart=always

# Use systemd’s ability to disable security-sensitive features
# that listmonk does not explicitly need.
# NoNewPrivileges should be enabled by DynamicUser=yes but systemd-analyze
# still recommended to explicitly enable it.
NoNewPrivileges=True
# listmonk doesn’t need any capabilities as defined by the linux kernel
# see: https://man7.org/linux/man-pages/man7/capabilities.7.html
CapabilityBoundingSet=
# listmonk only executes native code with no need for any other ABIs.
SystemCallArchitectures=native

# Make /home/, /root/, and /run/user/ inaccessible.
# If you set ExecStartPre=/usr/bin/mkdir -p "listmonk/uploads" to a directory in /home/ or /root/ it will cause uploads to fail 
# See https://github.com/knadh/listmonk/issues/843#issuecomment-1836023524
ProtectHome=True

# Make sure files created by listmonk are only readable by itself and
# others in the listmonk system group.
UMask=0027
# listmonk only needs to support the IPv4 and IPv6 address families.
RestrictAddressFamilies=AF_INET AF_INET6

[Install]
WantedBy=multi-user.target
```

Maybe `ExecStartPre=/usr/bin/mkdir -p "/listmonk/uploads"` should be changed to `/etc/listmonk/uploads`.

And still missing is a static directory. https://listmonk.app/docs/templating/#system-templates

**Comment by MaximilianKohler** [comment]
@kmohrf does everything look good now with this service and [the new simple.service](https://github.com/knadh/listmonk/pull/1622)? The only thing I'm not sure about is the `HOME` in `Environment=HOME=%S/listmonk-%i`, and if I broke something by changing `ExecStartPre=/usr/bin/mkdir -p "${HOME}/uploads"`.

**Comment by MaximilianKohler** [comment]
## Logs:

I'm not sure which is best, but when reviewing https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html I saw these options and added them commented out in the simple.service https://github.com/knadh/listmonk/blob/master/listmonk-simple.service file:
```
#LogsDirectory=/etc/listmonk
#CacheDirectory=/etc/listmonk/cache
```
I assume that listmonk doesn't use a cache, so I can probably remove that? I don't know how to get listmonk to save logs with that though, and this (alternative?) may be the easiest/best way (which is working for me):
```
ExecStart=/bin/bash -ce "exec /usr/bin/listmonk --config /etc/listmonk/config.toml --static-dir /etc/listmonk/static >>/etc/listmonk/listmonk.log 2>&1"
```
So I added it as an option. I'm not sure if it conflicts with anything else.

**Comment by MaximilianKohler** [comment]
I switched to AlmaLinux 9 and added back many of [the commands I previously removed due to them causing error messages](https://github.com/knadh/listmonk/issues/843#issuecomment-1836023524) because of an old systemd version.

```
systemctl --version
systemd 252
```

Running these consecutively:
```
systemctl daemon-reload
systemctl status listmonk
```
Still results in:
```
/etc/systemd/system/listmonk.service:54: Invalid device node path 'False', ignoring.
```
Not sure if that's the expected result though. Line 54 is this:
```
# listmonk doesn’t handle any specific device nodes.
DeviceAllow=False
```

File: listmonk@.service [hunk 1]
```diff
@@ -8,12 +11,23 @@ After=postgresql.service
 
 [Service]
 Type=simple
+PermissionsStartOnly=true
 EnvironmentFile=-/etc/default/listmonk
 EnvironmentFile=-/etc/default/listmonk-%i
-ExecStartPre=/usr/bin/mkdir -p "${HOME}/uploads"
+ExecStartPre=/usr/bin/mkdir -p "/etc/listmonk/uploads"
```
Comment (knadh) [inline listmonk@.service:17]: This shouldn't default to `/etc/listmonk` as a) the app should be run as a non-root user and hence, permission to write to /etc/ is unlikely b) arbitrary user files/content should not go in `/etc/*`.

Comment (MaximilianKohler) [reply listmonk@.service:17]: Where should they go, given that `${HOME}` is incompatible with `ProtectHome=True`?

Comment (knadh) [reply listmonk@.service:17]: Did a bit of Googling and looks like `BindPaths, ReadWritePaths` are the way to go. I am not familiar with it, so will have to experiment.

**Pull request closed without merging**